File: gluecodium/src/test/resources/smoke/comments/output/android/com/example/smoke/CtorLinks.java
/*
 *
 */
package com.example.smoke;
import android.support.annotation.NonNull;
import com.example.NativeBase;
public final class CtorLinks extends NativeBase {
    /**
     * <p>This class has just one constructor {@link com.example.smoke.CtorLinks.SingleCtor#SingleCtor()}.
     */
    public static final class SingleCtor extends NativeBase {
        public SingleCtor() {
            this(create(), (Object)null);
            cacheThisInstance();
        }
        /**
         * For internal use only.
         * @hidden
         * @param nativeHandle The SDK nativeHandle instance.
         * @param dummy The SDK dummy instance.
         */
        protected SingleCtor(final long nativeHandle, final Object dummy) {
            super(nativeHandle, new Disposer() {
                @Override
                public void disposeNative(long handle) {
                    disposeNativeHandle(handle);
                }
            });
        }
        private static native void disposeNativeHandle(long nativeHandle);
        private native void cacheThisInstance();
        private static native long create();
    }

    /**
     * <p>This class has just one constructor with one argument {@link com.example.smoke.CtorLinks.SingleCtorWithOneArgument#SingleCtorWithOneArgument(int)}.
     */
    public static final class SingleCtorWithOneArgument extends NativeBase {
        public SingleCtorWithOneArgument(final int arg) {
            this(create(arg), (Object)null);
            cacheThisInstance();
        }

        /**
         * For internal use only.
         * @hidden
         * @param nativeHandle The SDK nativeHandle instance.
         * @param dummy The SDK dummy instance.
         */
        protected SingleCtorWithOneArgument(final long nativeHandle, final Object dummy) {
            super(nativeHandle, new Disposer() {
                @Override
                public void disposeNative(long handle) {
                    disposeNativeHandle(handle);
                }
            });
        }

        private static native void disposeNativeHandle(long nativeHandle);
        private native void cacheThisInstance();
        private static native long create(final int arg);
    }

    /**
     * <p>This class has just one constructor with two argument {@link com.example.smoke.CtorLinks.SingleCtorWithTwoArgument#SingleCtorWithTwoArgument(int, String)}.
     */
    public static final class SingleCtorWithTwoArgument extends NativeBase {
        public SingleCtorWithTwoArgument(final int arg, @NonNull final String arg2) {
            this(create(arg, arg2), (Object)null);
            cacheThisInstance();
        }

        /**
         * For internal use only.
         * @hidden
         * @param nativeHandle The SDK nativeHandle instance.
         * @param dummy The SDK dummy instance.
         */
        protected SingleCtorWithTwoArgument(final long nativeHandle, final Object dummy) {
            super(nativeHandle, new Disposer() {
                @Override
                public void disposeNative(long handle) {
                    disposeNativeHandle(handle);
                }
            });
        }

        private static native void disposeNativeHandle(long nativeHandle);
        private native void cacheThisInstance();
        private static native long create(final int arg, @NonNull final String arg2);
    }

    public static final class OverloadedCtors extends NativeBase {
        public OverloadedCtors(@NonNull final String input) {
            this(create(input), (Object)null);
            cacheThisInstance();
        }
        /**
         *
         * @deprecated <p>Use {@link com.example.smoke.CtorLinks.OverloadedCtors#OverloadedCtors(String)} instead.
         * @param input
         * @param flag
         */
        @Deprecated
        public OverloadedCtors(@NonNull final String input, final boolean flag) {
            this(create(input, flag), (Object)null);
            cacheThisInstance();
        }
        /**
         * For internal use only.
         * @hidden
         * @param nativeHandle The SDK nativeHandle instance.
         * @param dummy The SDK dummy instance.
         */
        protected OverloadedCtors(final long nativeHandle, final Object dummy) {
            super(nativeHandle, new Disposer() {
                @Override
                public void disposeNative(long handle) {
                    disposeNativeHandle(handle);
                }
            });
        }
        private static native void disposeNativeHandle(long nativeHandle);
        private native void cacheThisInstance();
        private static native long create(@NonNull final String input);
        private static native long create(@NonNull final String input, final boolean flag);
    }
    /**
     * For internal use only.
     * @hidden
     * @param nativeHandle The SDK nativeHandle instance.
     * @param dummy The SDK dummy instance.
     */
    protected CtorLinks(final long nativeHandle, final Object dummy) {
        super(nativeHandle, new Disposer() {
            @Override
            public void disposeNative(long handle) {
                disposeNativeHandle(handle);
            }
        });
    }
    private static native void disposeNativeHandle(long nativeHandle);
}

File: gluecodium/src/test/resources/smoke/comments/output/android-kotlin/com/example/smoke/CtorLinks.kt
/*

 *
 */

@file:JvmName("CtorLinksExtensions")

package com.example.smoke

import com.example.NativeBase

class CtorLinks : NativeBase {

    /**
     * This class has just one constructor [com.example.smoke.CtorLinks.SingleCtor.SingleCtor].
     */
    class SingleCtor : NativeBase {




        constructor() : this(create(), null as Any?) {
            cacheThisInstance();
        }

        /**
         * For internal use only.
         * @suppress
         * @param nativeHandle The handle to resources on C++ side.
         * @param tag Tag used by callers to avoid overload resolution problems.
         */
        protected constructor(nativeHandle: Long, @Suppress("UNUSED_PARAMETER") tag: Any?)
            : super(nativeHandle, { disposeNativeHandle(it) }) {}

        private external fun cacheThisInstance()






        companion object {
            @JvmStatic private external fun disposeNativeHandle(nativeHandle: Long)

            @JvmStatic external fun create() : Long
        }
    }

    /**
     * This class has just one constructor with one argument [com.example.smoke.CtorLinks.SingleCtorWithOneArgument.SingleCtorWithOneArgument].
     */
    class SingleCtorWithOneArgument : NativeBase {




        constructor(arg: Int) : this(create(arg), null as Any?) {
            cacheThisInstance();
        }

        /**
         * For internal use only.
         * @suppress
         * @param nativeHandle The handle to resources on C++ side.
         * @param tag Tag used by callers to avoid overload resolution problems.
         */
        protected constructor(nativeHandle: Long, @Suppress("UNUSED_PARAMETER") tag: Any?)
            : super(nativeHandle, { disposeNativeHandle(it) }) {}

        private external fun cacheThisInstance()






        companion object {
            @JvmStatic private external fun disposeNativeHandle(nativeHandle: Long)

            @JvmStatic external fun create(arg: Int) : Long
        }
    }

    /**
     * This class has just one constructor with two argument [com.example.smoke.CtorLinks.SingleCtorWithTwoArgument.SingleCtorWithTwoArgument].
     */
    class SingleCtorWithTwoArgument : NativeBase {




        constructor(arg: Int, arg2: String) : this(create(arg, arg2), null as Any?) {
            cacheThisInstance();
        }

        /**
         * For internal use only.
         * @suppress
         * @param nativeHandle The handle to resources on C++ side.
         * @param tag Tag used by callers to avoid overload resolution problems.
         */
        protected constructor(nativeHandle: Long, @Suppress("UNUSED_PARAMETER") tag: Any?)
            : super(nativeHandle, { disposeNativeHandle(it) }) {}

        private external fun cacheThisInstance()






        companion object {
            @JvmStatic private external fun disposeNativeHandle(nativeHandle: Long)

            @JvmStatic external fun create(arg: Int, arg2: String) : Long
        }
    }

    class OverloadedCtors : NativeBase {




        constructor(input: String) : this(create(input), null as Any?) {
            cacheThisInstance();
        }
        /**
         *
         * @param input
         * @param flag
         */
        @Deprecated("Use [com.example.smoke.CtorLinks.OverloadedCtors.OverloadedCtors] instead.")

        constructor(input: String, flag: Boolean) : this(create(input, flag), null as Any?) {
            cacheThisInstance();
        }

        /**
         * For internal use only.
         * @suppress
         * @param nativeHandle The handle to resources on C++ side.
         * @param tag Tag used by callers to avoid overload resolution problems.
         */
        protected constructor(nativeHandle: Long, @Suppress("UNUSED_PARAMETER") tag: Any?)
            : super(nativeHandle, { disposeNativeHandle(it) }) {}

        private external fun cacheThisInstance()






        companion object {
            @JvmStatic private external fun disposeNativeHandle(nativeHandle: Long)

            @JvmStatic external fun create(input: String) : Long

            @JvmStatic external fun create(input: String, flag: Boolean) : Long
        }
    }



    /**
     * For internal use only.
     * @suppress
     * @param nativeHandle The handle to resources on C++ side.
     * @param tag Tag used by callers to avoid overload resolution problems.
     */
    protected constructor(nativeHandle: Long, @Suppress("UNUSED_PARAMETER") tag: Any?)
        : super(nativeHandle, { disposeNativeHandle(it) }) {}







    companion object {
        @JvmStatic private external fun disposeNativeHandle(nativeHandle: Long)
    }
}

